tutorials/shaders/shader_reference/shading_language.rst

```diff
@@ -448,42 +448,46 @@ Godot Shading language supports the most common types of flow control:
 
 .. code-block:: glsl
 
-    // if and else
+    // `if` and `else`.
     if (cond) {
 
     } else {
 
     }
 
-    // switch
+    // Ternary operator.
+    // This is an expression that behaves like `if`/`else` and returns the value.
+    // If `cond` evaluates to `true`, `result` will be `9`. Otherwise, `result` will be `5`.
+    int result = cond ? 9 : 5;
+
+    // `switch`.
     switch (i) { // signed integer expression
         case -1:
             break;
         case 0:
             return; // break or return
-        case 1: // pass-through
+        case 1: // fall-through
         case 2:
             break;
         //...
         default: // optional
             break;
     }
 
-    // for loops
+    // `for` loop. Best used when the number of elements to iterate on is known in advance.
     for (int i = 0; i < 10; i++) {
 
     }
 
-    // while
-    while (true) {
+    // `while` loop. Best used when the number of elements to iterate on is not known in advance.
+    while (cond) {
 
     }
 
-    // do while
-    // Like `while`, but always runs at least once.
+    // `do while`. Like `while`, but always runs at least once.
```

Comment:
I don't really understand how while and do while differ from this could we have example of code for both illustrating a difference